components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -98,9 +98,48 @@ const XminHorizonGuidance: React.FunctionComponent<CheckGuidanceProps> = ({
   );
 };
 
+type HeldBackInfoType = {
+  xmin: number;
+  assigned_at: number;
+}
+
+function epochFromFullTransactionId(value: number): number {
+  return Number(BigInt(value) >> BigInt(32));
+}
+
+function xidFromFullTransactionId(value: number | null): number | null {
+  if (value == null) {
+    return null;
+  }
+  const epoch = epochFromFullTransactionId(value);
+  return fullTransactionIdMinusEpoch(value, epoch);
+}
+
+function fullTransactionIdMinusEpoch(
+  value: number,
+  epoch: number,
+): number {
+  return Number(BigInt(value) - (BigInt(epoch) << BigInt(32)));
+}
```

Comment:
These functions are from `app/javascript/utils/txid.ts` in the main app, but I don't see a good way to share these (since this is also included in the public website code), thus copied them.

Reply:
Given the limited amount of code and the fact that it won't change much, I think this is okay. Maybe we should add a comment referencing the source? If you think it's not appropriate to reference the private code in the public docs, I'm okay to leave as-is.

Reply:
I was going to suggest to add `xmin_txt` or something as a new hash value so that xmin will keep xid8 whereas `xmin_txt` will have `${epoch}:${xid}` (then we don't need to add a duplicated xid logic here).
Then, I sadly realized that we actaully don't have the logic to make `xmin_txt` in our backend handy (it's not so difficult to get `xid` part, we could do `xmin_horizon_backend::text::xid` during SELECT, but I don't think there is a handy way to get an epoch) so this code needs to be added somewhere newly :/
With that, I'd say that it's okay to add it here. Re: comment, how about adding it in the private repo side?